### ISSUE TITLE
Powersave text is too blurry in landscape mode.

### ISSUE THREAD

**Issue by 133794m3r** [00-Evan/shattered-pixel-dungeon#128]
The text is blurry(I have always used "use smooth font") and yeah I know it's suppsoede to be a bit blurrier but on my galaxy nexus the font is literally barely ledigble both phones are 1280x720(in landscape) one is an amoled pentile display, which I know is a lower quality display.

I think that you need some sharpening filtering applied to it, it looks like billinear filtering is being applied by the blurriness and it's unusable on one phone the other one is OK still a bit blurry in text but overall OK.

I can provide screenshots if you need, I'll be using it on one as text is kinda important with it being a roguelike. 

Portrait for both seems fine ATM text is blurry but not horrible like any normal billinear filtered image but portrait is gets a lot worse.

**Comment by 00-Evan** [comment]
I have no control over the filtering used in power saver mode. If I did, I would use nearest-neighbor wherever possible.

As we discussed before, your device has different scaling limits depending on whether it is in portrait or landscape mode. 5x vs 4x. The game attempts to run at half scale in power saver, rounded up, with a minimum of 2x, so in this case it would run at 3x in portrait and 2x in landscape. This is what's leading to the increased blurriness in landscape.

In the specific case of a 4x display I can see how this would result in a lot of blurriness. The game should probably target 3x in power saver in those cases.

Rather than round(0.5x), round(0.4x + 1) would work better.

**Comment by 133794m3r** [comment]
Wait, I don't remember you saying you ahd no control over what scaling was done, I know you said you put in the spirtes and get a frame back, is the scaling all done via android then?

**Comment by 133794m3r** [comment]
Wait, I don't remember you saying you ahd no control over what scaling was done, I know you said you put in the spirtes and get a frame back, is the scaling all done via android then?

Also it's the pentile display that it's ltierally impossible to read, the normal screen it's readable blurry but readable I think it's the combo of the old pentile display 2 subpixels per pixel instead of 3 and the 2x rendering resolution.  I don't know what devices would be using as 4x rendering mode natively offhand but I'm going to be doing a few run-throughs and see if I can douplicate any of the bugs I was about to report before 0.4.3 came out.

P.S. One was items showing up as "NULL" in all-red and I had a couple of crashes but only once or twice and I couldn't lock it down.

**Comment by 00-Evan** [comment]
The scaling uses the built-in android hardware scaler. I could manually scale using openGL and use whatever filtering I want, but this would not utilize the same optimized scaler, and would thus have a large performance impact, negating a fair chunk of the benefit of power saver in the first place. 

It's also worth noting that the bilinear filtering looks very good in most cases, your specific situation of 4x to 2x is easily the worse case for it, and is why i'm adjusting the formula to have it render 3x in this case. As an example of it working well, on high-end phones with 1440p displays, the game renders at around 720p and the difference is extremely difficult to notice, yet the game requires about 30-40% the graphical horsepower.

I've already run  some tests using an emulated device and the difference between 2x and 3x is very significant
2x:
![2x](https://cloud.githubusercontent.com/assets/2347603/19539708/daa17b52-9628-11e6-8b37-9c907aae2646.png)

3x:
![3x](https://cloud.githubusercontent.com/assets/2347603/19539709/dd19ff80-9628-11e6-89b2-a21f9298a3af.png)

If you've got any other bugs to report I'll be happy to look into them, though please post them as a separate issue. Someone also also mentioned a 'null' item to me, so there definitely is something to that one.

**Comment by 133794m3r** [comment]
Wel as I said I had it, before I saw the update(f-droid updated repos and told me there was an update). Do you normally log via alogcat? Otherwise I don't know how to provide you the save file to do tests with as my phone I had the issue on was non-rooted locked bootloader. For what it's worth, it was apparently a dew droplet that one time. And I will, I'll ry tobu gtest this thing.

As I had said, thefiltering was fine in whatever it was doing(I guess 3x) in portrait but with pentile display of the old/cheaper type it really made it bad.

The 2x screen was about the same difference between 2x and 3x from the emulator again applied to 2x in terms of unreadableness on the old nexus Ihave.

**Comment by 133794m3r** [comment]
Do you have any way to enable debugging fro the "release" branches? I got the null item bug but after reloading(exiting app and returning) itwas switched with the proper icon and there was no logging to alogcat.

**Comment by 00-Evan** [comment]
I don't do much logging, regardless this likely isn't a save data related issue. 

If you're able to reproduce it and can get screen shots or other details, please post those as a new issue.

**Comment by 00-Evan** [comment]
Changes implemented in commit a5d835f2f41887c61a6394e4a61e743c0672ab6c
